### ISSUE TITLE
need help implementing a plugin

### ISSUE THREAD

**Issue by stefanos-kalantzis** [spotbugs/spotbugs#1282]
I need to write a plugin that will run a check on a specific method when that is called.

More precisely:
- `Instant.now().truncatedTo(MICROS);` :heavy_check_mark: 
- `Instant.now();` :heavy_multiplication_x: 

I don't want to allow nanosecond precision for the `now()` method of Instant.

Is this possible with spotbugs?
If so, any starter hints would be greatly appreciated.
I skimmed the docs, but could find anything useful.

Thanks!

---

The reason for this is that we are upgrading a product that is currently running in Java 8.
And because of https://bugs.openjdk.java.net/browse/JDK-8068730, many of our dates are now generated with nanosecond precision. Which causes testing issues because our persistence layer (MariaDB) doesn't support nanosecond precision, but only microsecond precision.
I know there are other solutions to this problem, but I want to explore all possibilities.

**Comment by ThrawnCA** [comment]
I'm sure that it is possible, but offhand I couldn't advise how. What I would suggest is taking a look at the detector that forbids calls to `System.exit` (DM_EXIT) for a better understanding of how to detect calls to a specific method, and working from there.

**Comment by stefanos-kalantzis** [comment]
Thanks for the reply. I actually forgot to update this, but I managed to implement this based on some examples that I found in `fb-contrib`. I didn't notice the `System.exit` one, that would probably help me even more.

I don't think it makes sense to post here my code since it's very unlikely that someone will stumble upon this.
If anyone thinks otherwise I'll happily post it :)

For now I'm closing this.
